### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by alvaroiporre**
## Project requirements
In this PR we added tests for all the methods in the following classes:

 - Book
 - Person
 - Student
 - Teacher
 - Classroom
 - Rental

### CODE REVIEW

File: spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      classroom = Classroom.new(classroom_label)
+      expect(classroom.label).to eq(classroom_label)
+    end
+
+    it 'initializes an empty array of students' do
+      classroom = Classroom.new(classroom_label)
+      expect(classroom.students).to be_empty
+    end
+  end
+end
```
Comment (Gambit142) [inline spec/classroom_spec.rb]: - [ ] One of the major requirements for this milestone is to test all methods in all classes. Kindly write a test for the `add_student` method in the `Classroom` class. Testing methods in classes help in identifying bugs and errors early in the development cycle, saving time and money. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @alvaroiporre @theresetuyi 👋, 

Good job so far!

## Highlights
- All linter checks passed ✔️ 
- Installed `Rspec` library ✔️ 
- All test files are organized in a spec folder ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️

- [ ] One of the major requirements for this milestone is to write unit tests for all methods in all classes. Kindly write tests for all methods in the following classes:

- _**CapitalizeDecorator**_
- _**Decorator**_ 
- _**Nameable**_
- _**TrimmerDecorator**_

image

### _Kindly check the comments under the review for other required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/classroom_spec.rb (hunk 1 above)
Comment (cynthiainga) [inline spec/classroom_spec.rb]: - [ ] Good job so far, but as mentioned by the previous reviewer you should add the test for the `add_student` method in the `classroom` class. Please kindly 🙃 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @alvaroiporre & @theresetuyi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [ ] Installed Rspec library ✔️
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comment under the review 👇🏾 👇🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @alvaroiporre & @theresetuyi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All tests are passing ✔️ 
- [ ] Installed Rspec library ✔️
- [ ] No linters errors ✔️
- [ ] Correct Gitflow ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**